src/client/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from 'react';
+import Product from '../Product/Product.component';
+import './CarouselStyle.css';
+
+
+export default function Carousel({ products, show }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [length, setLength] = useState(products.length);
+  const [touchPosition, setTouchPosition] = useState(null);
+
+  useEffect(() => {
+    setLength(products.length);
+  }, [products]);
+
+  const next = () => {
+    if (currentIndex < length - show) {
+      setCurrentIndex((prevState) => prevState + 1);
+    }
+  };
+
+  const prev = () => {
+    if (currentIndex > 0) {
+      setCurrentIndex((prevState) => prevState - 1);
+    }
+  };
+
+  const handleTouchStart = (e) => {
+    const touchDown = e.touches[0].clientX;
+    setTouchPosition(touchDown);
+  };
+
+  const handleTouchMove = (e) => {
```

Comment:
This seems to be buggy, let's remove it for now. You can fix it in a new PR if you want, touch support isn't important for now.